### DISCUSSION TITLE
Prisma VS TypeORM schema generation

### DISCUSSION THREAD

**Discussion by orenmizr** [prisma/prisma#11605] [Q&A] [answered]
in **TypeORM** you can define decorators on the model to generate the database entity. 

in **Prisma** you have the prisma schema  (with data source... ) which is not auto generated and has it own SDL so you can't really have your models be dynamically generated and injected to it. 

i know there is a code first with nexus, but i think (opinion) it looks nasty and unreadable. how can i have my prisma schema generated by decorators like typeORM ?

**Comment by pantharshit00** [comment, marked as answer]
Hello,

Right now there is no official option of creating a Prisma schema via code. You can follow: https://github.com/prisma/prisma/issues/227

Regarding graphql usage, there is typegraphql prisma generator which can ease the interop between them: https://typegraphql.com/docs/prisma.html

**Comment by mohammadsaadshafiq** [comment]
Hi, I wrote this generic article for future reference https://medium.com/p/176f28b04933

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
